### PR TITLE
Feat/sc 145 category range cards

### PR DESCRIPTION
**Pull request by davidgilbertson**
Jira: https://govausites.atlassian.net/browse/SC-145

Adds a chart to the indicator cards on the all-cities pages, with miscellaneous other design tweaks around the cards.

### CODE REVIEW

File: src/components/CategoryOverview/CategoryOverview.js [hunk 1]
```diff
@@ -21,14 +20,6 @@ const classnames = require('classnames/bind').bind(style);
...
   const isContextCategory = props.category.id === CATEGORY_IDS.CONTEXT;
 
-  const indicators = props.category.overviewIndicatorIds.map((indicatorId) => {
-    const value = props.city
-      ? props.city.indicators[indicatorId]
-      : getMinAndMaxForIndicator(indicatorId, props.cities);
-
-    return { id: indicatorId, value };
-  });
-
```
Comment (davidgilbertson) [inline src/components/CategoryOverview/CategoryOverview.js:31]: Previously data was aggregated before being sent down to the card. Now the 'range card' handles it.

File: src/components/AllCitiesOverviewContainer/AllCitiesOverview/AllCitiesOverview.js [hunk 2]
```diff
@@ -103,11 +102,6 @@ const AllCitiesOverview = props => (
       />
     </div>
 
-    <PageLegend
-      colorName="primary"
-      isAllCitiesPage
-    />
```
Comment (davidgilbertson) [inline src/components/AllCitiesOverviewContainer/AllCitiesOverview/AllCitiesOverview.js:109]: No such thing as a `<PageLegend>` any more

File: src/components/CityColumnChart/CityColumnChart.js [hunk 3]
```diff
@@ -365,21 +364,11 @@ class CityColumnChart extends Component {
...
-                icon={firstIndicator.contextual ? 'contextualIndicator' : 'performanceIndicator'}
-                size={14}
-                title=""
-              />
-            </Tooltip>
+            <IndicatorIcon
+              className={style.indicatorTypeMark}
+              color={getColorVariant(props.colorBase, '500')}
+              contextual={firstIndicator.contextual}
+            />
```
Comment (davidgilbertson) [inline src/components/CityColumnChart/CityColumnChart.js:371]: There were no visible changes to the chart, this just abstracts the "C" or "P" icon wrapped in a tooltip to a new `IndicatorIcon` component.

File: src/components/IndicatorCard/IndicatorCard.js [hunk 4]
```diff
@@ -1,153 +1,42 @@
 import React from 'react';
```
Comment (davidgilbertson) [inline src/components/IndicatorCard/IndicatorCard.js:1]: Previously there was an `IndicatorCard` that showed either a single number (on the cities page) or two numbers (on the all cities pages).

Now the two cards are sufficiently different that they are two different components. However, we still use `<IndicatorCard>`, and it just selects which card to show.

File: src/components/IndicatorCard/IndicatorCard.scss [hunk 5]
```diff
@@ -1,168 +1,9 @@
...
-    margin-bottom: 4px;
-  }
-
-  .largeCard.dualNumber .firstNumber & {
-    font-size: 24px;
-  }
-
-  .largeCard.dualNumber .secondNumber & {
-    font-size: 18px;
-  }
```
Comment (davidgilbertson) [inline src/components/IndicatorCard/IndicatorCard.scss:159]: So much deleted code :(

File: src/components/IndicatorCard/NumberCard/NumberCard.scss [hunk 6]
```diff
@@ -0,0 +1,107 @@
+@import '~style/variables';
```
Comment (davidgilbertson) [inline src/components/IndicatorCard/NumberCard/NumberCard.scss:1]: `NumberCard` is mostly copy/pasted over from `IndicatorCard` (there should be almost no change to the indicator cards on the city pages).

But a lot of the branching logic is gone now.

File: src/components/IndicatorCard/RangeCard/RangeCard.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                y={`${(1 - height) * 100}%`}
+                width={`${barWidth * 100}%`}
+                height="200%"
+                rx="2"
+                ry="2"
+                fill={fill}
+              />
+            );
+          })}
+        </svg>
```
Comment (davidgilbertson) [inline src/components/IndicatorCard/RangeCard/RangeCard.js:73]: I considered making this little chart a separate component. Reasons I didn't:
- it's not reused elsewhere
- it's < 50 lines
- it shares quite a bit with other parts of this component

File: src/components/IndicatorCard/RangeCard/RangeCard.scss [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  .smallCard & {
+    top: 2px;
+    left: 4px;
+  }
+
+  .largeCard & {
+    top: 4px;
+    left: 9px;
+  }
+}
```
Comment (davidgilbertson) [inline src/components/IndicatorCard/RangeCard/RangeCard.scss:56]: There is some duplication here between the `RangeCard` and the `NumberCard`, but a lot of stuff is _slightly_ different. A little bit of repetition is better than a lottle bit of complexity.

File: src/components/PageLegend/PageLegend.test.js [hunk 9]
```diff
@@ -1,62 +0,0 @@
...
-it('should not show the range icons on a city page', () => {
-  const component = shallow(
-    <PageLegend
-      {...defaultProps}
-      isAllCitiesPage={false}
-    />
-  );
-
-  expect(component.find('.rangeWrapper').length).toBe(0);
-});
```
Comment (davidgilbertson) [inline src/components/PageLegend/PageLegend.test.js:62]: 277 lines of code, barely a month old. Deleted. Gone. Kaput. :(

File: src/helpers/getMinAndMaxForIndicator.test.js [hunk 10]
```diff
@@ -1,44 +0,0 @@
...
-  const result = getMinAndMaxForIndicator('population', citiesMock);
-
-  expect(result).toEqual([50, 100]);
-});
-
-it('should return the min and max including negatives', () => {
-  const result = getMinAndMaxForIndicator('unemploymentGrowth', citiesMock);
-
-  expect(result).toEqual([-0.08, 0.07]);
-});
```
Comment (davidgilbertson) [inline src/helpers/getMinAndMaxForIndicator.test.js:44]: More deleted code :(

File: src/components/IndicatorCard/RangeCard/RangeCard.scss (hunk 8 above)
Comment (jonalport) [reply src/components/IndicatorCard/RangeCard/RangeCard.scss:56]: You say a little, I say a lottle - what you need is a little bot.

File: src/components/IndicatorCard/RangeCard/RangeCard.js (hunk 7 above)
Comment (jonalport) [reply src/components/IndicatorCard/RangeCard/RangeCard.js:73]: No reuse is a good reason I feel.

**Review by jonalport — APPROVED** [review]
Range cards look good!

**Comment by jonalport** [comment]
I have duplicated some of these changes in the branch I'm working on. 

This is gonna be a fun merge O_O

**Pull request merged**